### PR TITLE
[DependencyInjection] Exclude referencing service (self) in `TaggedIterator`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Closes https://github.com/symfony/symfony-docs/issues/17656

### CODE REVIEW

File: service_container/tags.rst [hunk 1]
```diff
@@ -692,6 +692,11 @@ iterator, add the ``exclude`` option:
 
     The ``exclude`` option was introduced in Symfony 6.1.
 
+.. versionadded:: 6.3
```
Comment (alamirault) [inline service_container/tags.rst:700]: When tagging 7.0 all versionadded directives will be drop, so maybe it should be a note/tip + versionadded ?

Comment (OskarStark) [reply service_container/tags.rst:700]: Yes that would be better 👍

Comment (alexandre-daubois) [reply service_container/tags.rst:700]: That was my main concern about this PR. Updated it with a note+versionadded, thank you both!

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Alexandre.

**Pull request merged**